core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java

```diff
@@ -236,26 +236,29 @@ else if (element instanceof LikeClause) {
                 throw new TrinoException(GENERIC_INTERNAL_ERROR, "Invalid TableElement: " + element.getClass().getName());
             }
         }
-
-        Map<String, Expression> sqlProperties = mapFromProperties(statement.getProperties());
+        if (hasDefaultProperty(statement.getProperties())) {
+            throw semanticException(NOT_SUPPORTED, statement, "Setting a table property to DEFAULT in a CREATE TABLE statement is not supported");
+        }
         Map<String, Object> properties = metadata.getTablePropertyManager().getProperties(
                 catalogName,
                 tableName.getCatalogName(),
-                sqlProperties,
+                statement.getProperties(),
                 session,
                 metadata,
                 accessControl,
-                parameterLookup,
-                true);
+                parameterLookup);
 
-        if (!disableSetPropertiesSecurityCheckForCreateDdl && !properties.isEmpty()) {
```

Comment:
nit: remove this check as separate commit, so that we can revert it alone (or point to it) if someone complains because access control stopped working


We can leave check for now too.

Reply:
I have extracted that change to a separate commit.